tests/losses/test_aucm_loss.py

```
@@ -0,0 +1,116 @@
+EXPECTED_V1 = 1.25
+EXPECTED_V2 = 5.0
+
+
+class TestAUCMLoss(unittest.TestCase):
+    """Test cases for AUCMLoss."""
+
+    @parameterized.expand([("v1",), ("v2",)])
+    def test_versions(self, version):
+        """Test AUCMLoss with different versions."""
+        loss_fn = AUCMLoss(version=version)
+        input = torch.randn(32, 1, requires_grad=True)
```

Ruff (0.15.7)

[error] Variable `input` is shadowing a Python builtin

(A001)